packages/gatsby/src/cache-dir/component-renderer.js

```diff
@@ -67,11 +77,16 @@ class ComponentRenderer extends React.Component {
 
   render() {
     if (this.state.pageResources) {
-      return createElement(this.state.pageResources.component, {
-        key: this.props.location.pathname,
-        ...this.props,
-        ...this.state.pageResources.json,
-      })
+      return createElement(
```

Comment:
This doesn't work as it turns out because it means layouts are re-rendered on every page change which breaks the contract that they're stateful which breaks among other things gatsbygram.

Also it's less performant.

Thinking about alternatives...